Armadillo/src/main/java/com/scribd/armadillo/playback/PlaybackEngine.kt

```diff
@@ -137,16 +140,21 @@ internal class ExoplayerPlaybackEngine(private var audioPlayable: AudioPlayable)
 
         exoPlayer = createExoplayerInstance(context, audioAttributes.exoPlayerAttrs, loadControl)
 
-        val mediaSource = mediaSourceRetriever.generateMediaSource(audioPlayable.request, context)
-        exoPlayer.setMediaSource(mediaSource)
-        exoPlayer.prepare()
+        try {
+            val mediaSource = mediaSourceRetriever.generateMediaSource(audioPlayable.request, context)
+            exoPlayer.setMediaSource(mediaSource)
+            exoPlayer.prepare()
 
-        exoPlayer.addListener(playerEventListener)
+            exoPlayer.addListener(playerEventListener)
 
-        exoPlayer.playWhenReady = config.isAutoPlay
+            exoPlayer.playWhenReady = config.isAutoPlay
 
-        stateModifier.dispatch(PlaybackEngineReady(true))
-        stateModifier.dispatch(PlayerStateAction(PlaybackState.PAUSED))
+            stateModifier.dispatch(PlaybackEngineReady(true))
+            stateModifier.dispatch(PlayerStateAction(PlaybackState.PAUSED))
```

Comment:
Definitely out of scope, but shouldn't the playback state be play/pause based on `playWhenReady`/`isAutoPlay`?

Reply:
It should update elsewhere